src/goto-programs/json_goto_trace.cpp

```diff
@@ -114,6 +111,13 @@ void convert(
             type_string=from_type(ns, identifier, symbol->type);
 
           json_assignment["mode"]=json_stringt(id2string(symbol->mode));
+          if(step.full_lhs_value.is_not_nil())
+            full_lhs_value=json(step.full_lhs_value, ns, symbol->mode);
```

Comment:
I think the `json` function does not handle that case, maybe it should but the type may be different: for now it returns a `json_objectt` while for `nil_exprt` the expected result would probably be `null_json_object`